Switch supported FreeBSD to 13.0

SeanTAllen opened the pull request.

SeanTAllen commented:
This should fail until the issues exposed in #159 are fixed.

Theodus commented on .cirrus.yml:
```diff
@@ -21,6 +21,24 @@ task:
+  only_if: $CIRRUS_PR != ''
+
+  freebsd_instance:
+    image: freebsd-12-1-release-amd64
```
We should use a FreeBSD 12.2 image for this.

```suggestion
    image: freebsd-12-2-release-amd64
```

SeanTAllen replied:
We are testing on 13 now so we should switch to it (see changes in ponyc repo and corral)

Theodus commented on .cirrus.yml:
```diff
@@ -21,6 +21,24 @@ task:
+  freebsd_instance:
+    image: freebsd-12-1-release-amd64
+
+  name: "PR: Test bootstrapping on FreeBSD 12.1"
```
```suggestion
  name: "PR: Test bootstrapping on FreeBSD 12.2"
```

SeanTAllen commented:
@Theodus do you want to update to FreeBSD 13 or should I?

Theodus commented:
> @Theodus do you want to update to FreeBSD 13 or should I?

I'll give it a shot.

ponylang-main commented:
<!-- release-note-reminder-bot -->
Hi @SeanTAllen,

The changelog - changed label was added to this pull request; all PRs with a changelog label need to have release notes included as part of the PR. If you haven't added release notes already, please do.

Release notes are added by creating a uniquely named file in the `.release-notes` directory. We suggest you call the file `161.md` to match the number of this pull request.

The basic format of the release notes (using markdown) should be:

```
## Title

End user description of changes, why it's important,
problems it solves etc.

If a breaking change, make sure to include 1 or more
examples what code would look like prior to this change
and how to update it to work after this change.
```

Thanks.

SeanTAllen commented:
@Theodus as this changes the supported FreeBSD, I changed the PR title and added a changelog. It will need some release notes as well. I'll add the release notes shortly.

Theodus commented:
@SeanTAllen this should be good to go now. Feel free to merge if you agree with the changes I made.
